### ISSUE TITLE
Metrics for GitRepo status

### ISSUE THREAD

**Issue by pcallewaert** [rancher/fleet#247]
I'm setting up a POC internally with fleet and k3s to deploy helm charts to edge devices. It works pretty good, but unfortunately I am currently missing feedback if the sync is successful. For example, one of the devices had a firewall issue to connect to GitHub.
I was wondering how this is mostly monitored? I was hoping to find a prometheus endpoint that exposed the status of a GitRepo config.
Also, I noticed by browsing through some issues that you can visualize some information with rancher, but currently I did not add this to our stack.

Thanks!

**Comment by ADustyOldMuffin** [comment]
I would also like this as a feature, and would be interested in taking this on as we currently need it.

**Comment by atsai1220** [comment]
Having Prometheus metrics would be incredibly useful when managing multiple downstream clusters. I would like Alertmanager to send Slack alerts when a specific bundle has been in a `modified` state for over x minutes with the specific cluster name.

It's true that downstream clusters with Prometheus and Alertmanager should be able to alert on unhealthy pods but there is no good way to determine if the bundle is `missing` or has `modified` resources unless determined upstream (from the Rancher console cluster).

**Comment by strowi** [comment]
Same here. previously we've always been using some ci/cd solution to actively deploy to kubernetes. An error in e.g. gitlab pipeline will notify the user.
This doesn't seem to be the way with fleet. An error in a deployment yaml would not surface anywhere.
Maybe it would also be a good idea to have fleet notify directly via mail or webhook since prometheus will always have some delay?

**Comment by ADustyOldMuffin** [comment]
@strowi you can use `kubectl wait` to see if it was successful which is what we use.

We wait for X amount of time and then use `kubectl wait` and check for the Ready status with a timeout.

**Comment by strowi** [comment]
@ADustyOldMuffin Thank you for the suggestion, will check that too. To be sure you're talking about the fleet-cli right?
I came here initially from using the rancher-ui continuous-delivery part, which just connects a repo with a cluster and does monitor this passively (much like argocd i guess).

**Comment by ADustyOldMuffin** [comment]
Nope, if you're using GitOps you could have your pipeline trigger on commit and then wait on the git repo to be in a ready state or the bundles if you know the bundles.

**Comment by strowi** [comment]
Right, that might be possible. Need to test and check that out a little more in context of multi-cluster and k8s-context.
I don't want to expose the credentials to a ci/cd-system outside of rancher (possibly vs. rancher pipeline).

**Comment by marthydavid** [comment]
ping we would like to get metrics from fleet

**Comment by manno** [comment]
Closing in favor of https://github.com/rancher/fleet/issues/1408
